modules/currency-l1/src/main/scala/org/tessellation/currency/l1/modules/Storages.scala

```diff
@@ -43,7 +46,9 @@ object Storages {
       globalL0ClusterStorage <- L0ClusterStorage.make[F](globalL0Peer)
       lastCurrencySnapshotStorage <- LastSnapshotStorage.make[F, S, SI]
       lastGlobalSnapshotStorage <- LastSnapshotStorage.make[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo]
-      transactionStorage <- TransactionStorage.make[F, T]
+      transactionStorage <- currencyIdentifier.value.value.hashF.map(TransactionReference.emptyCurrency).flatMap {
```

Comment:
Seems like this functionality repeats in several places (hash and build last tx ref) maybe it's better to extract it next to the `emptyCurrency` and use wherever needed?

Reply:
:+1: